components/org.wso2.carbon.micro.integrator.security/src/main/java/org/wso2/carbon/micro/integrator/security/vault/VaultTool.java

```diff
@@ -116,13 +124,20 @@ private static void printHelp() {
      * @param cipher cipher
      */
     private static String encryptedValue(Cipher cipher) {
+        addSystemProperties();
         String firstPassword;
         String secondPassword;
         while (true) {
             firstPassword = Utils.getValueFromConsole("Enter Plain Text Value : ", true);
             secondPassword = Utils.getValueFromConsole("Please Enter Value Again : ", true);
             if (!firstPassword.isEmpty() && firstPassword.equals(secondPassword)) {
-                if (firstPassword.length() < 5 || firstPassword.length() > 30) {
+                String passwordRegex = System.getProperty(Constants.SECURE_VALULT_PASSWORD_REGEX_SYSTEM_PROPERTY);
+
+                if (passwordRegex != null && !Pattern.matches(passwordRegex, firstPassword)) {
+                    System.out.println("Password must validates with the given regular expression" +
```

Comment:
Let's print the regex as well. Also grammar needs to be corrected. May be "Password does not match the given regular expression"

Reply:
Done